### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #20 [February] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#10120] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, February 8 (9am PDT/6pm CEST)**
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/10079
- Next: https://github.com/celo-org/celo-monorepo/discussions/10157
- Recording: [Google Drive](https://drive.google.com/file/d/1OxfXHLChNviTFhEBvHatV1337OdANuul/view?usp=sharing)

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [x] Encourage additions to the agenda
- [x] Publicise on forum and Discord
- [x] RECORD meeting 🎥 
- [x] Share recording below and on forum
- [x] Create next meeting (link to next meeting above)

**Comment by arthurgousset** [comment]
**Faucet page: [faucet.celo.org](https://faucet.celo.org/)**

TLDR:
- larger faucet amounts: 5 CELO, 5 cUSD, 5 cEUR, & 5 cREAL
- large requests: linked [Google form](https://docs.google.com/forms/d/e/1FAIpQLSfpt3WikYt5-TsDHmUgfFCbZjmZMcWr9bO5H0csHcmMrl9sFw/viewform)

<img width="600" alt="image" src="https://user-images.githubusercontent.com/46296830/217578777-4fcdda30-7d74-4280-b2d1-58f9448536aa.png">

Ref discussion: 
- https://github.com/celo-org/celo-monorepo/discussions/9957

**Comment by arthurgousset** [comment]
**WalletConnect v2**: 

TLDR: 
- we're upgrading [@celo/react-celo](https://www.npmjs.com/package/@celo/react-celo) to WalletConnect v2
  - beta release: Thursday, Feb 9
  - production release: Monday, Feb 13
 
Question:
- How are wallets doing with WalletConnect v2? (any feedback on integration)
- How are dapps doing with WalletConnect v2? (any feedback on integration)

Ref Github issue: 
- https://github.com/celo-org/react-celo/issues/341

**Reply by arthurgousset** [reply]
Meeting notes: 
- Worry that dapps won't be upgrading to WalletConnect v2
- Maybe pressing is communicating with dapps that use react-celo that they need to bump and deploy.
- about walletconnect v2, is there updates to the celo/rainbowkit-celo library as well?
- Native mobile dapps are not well supported in WCv2 (i.e. to connect another wallet)
  - https://github.com/WalletConnect/walletconnect-monorepo/issues/1118
- Query whether rainbowkit already supports WCv2 in v0.8.1
  - https://github.com/rainbow-me/rainbowkit/pull/1003
- I think I saw a setting in wagmi / rainbowkit to enable WC v2
  - here's the WC v2 setting in wagmi: https://wagmi.sh/react/connectors/walletConnect#version-optional
- [ ] Make a post in Celo Founders Telegram group
- [ ] Make a Twitter post about WalletConnect v2
- [ ] DevRel to point to WalletConnectv2 migration post

**Comment by arthurgousset** [comment]
**SocialConnect** (formerly "ASv2")

TLDR: 
- Node wallet, backend integration with NodeJS (`identitykit`) (demo and docs currently private)
  - docs: https://docs.nodefinance.org/category/identity
  - demo: https://youtu.be/abSI_C1SYqg
- Kaala integration, backend integration with NodeJS
- React Native demo app: [therealharpaljadeja](https://github.com/therealharpaljadeja)/[socialconnect_rn_demo](https://github.com/therealharpaljadeja/socialconnect_rn_demo) (tidying up atm)
- Web demo app: celo-org/[emisianto](https://github.com/celo-org/emisianto)
- NodeJS demo app: celo-org/[SocialConnect](https://github.com/celo-org/SocialConnect) (with Ethers, Web3, Contrackit)

Question: 
- How is Valora integration going? 
  - any issues with [resolve-kit](https://github.com/valora-inc/resolve-kit)?
  - https://github.com/valora-inc/resolve-kit/pull/143

New branding: 
<img width="500" alt="image" src="https://user-images.githubusercontent.com/46296830/217580481-58b7cf23-eff7-4396-b5e3-815642d8b5b0.png">

**Comment by arthurgousset** [comment]
**Ethersjs transaction type investigation** (@aaronmgdr)

Provide some context and early design ideas.

Ref:
- https://github.com/ethers-io/ethers.js/issues/3747
- https://github.com/celo-org/celo-monorepo/discussions/9922

**Reply by arthurgousset** [reply]
Meeting notes: 

> we are not alone -- zksync and other networks would benefit from this as well

**Comment by arthurgousset** [comment]
[@juanjgiraldoc] **gas fee currency in Metamask**: Where can I use this feature?
> I want to see this feature implemented in some wallets in the near future.
> I just imagine when I am going to approve a transaction I have the possibility to select in which token I want to pay the gas. $Celo or one of the Celo Stables

- cLabs team working on a Metamask snap
- cLabs team investigating adding new transaction type in Ethersjs

**Reply by 0xj4an-Work** [reply]
Thanks for your attention @0xarthurxyz

**Reply by aaronmgdr** [reply]
For context on the metamask snap https://forum.celo.org/t/h1-2023-primitives-team-roadmap/5167

**Comment by mattdkim** [comment]
Webhooks for Celo | Intro, status, feedback

**Reply by arthurgousset** [reply]
Meeting notes: 
- Building webhooks that users can configure to track blockchain state changes. 
  - See forum post: https://forum.celo.org/t/h1-2023-data-services-team-roadmap/5166
- Great for push notifications

**Comment by arthurgousset** [comment]
(for future reference) Here is the [recording](https://drive.google.com/file/d/1OxfXHLChNviTFhEBvHatV1337OdANuul/view?usp=sharing)
